test/solver.jl

```diff
@@ -320,6 +321,11 @@ k = 1
         @test size(d_F) == size(A)
         @test size(d_F.Q, 1) == size(A, 1)
         @test det(d_F.Q) ≈ det(collect(d_F.Q * CuMatrix{elty}(I, size(d_F.Q)))) atol=tol*norm(A)
+        @test d_F.Q[1, 1] ≈ qra.Q[1, 1]
```

Comment:
Why should this succeed without requiring the user to `@allowscalar` itself? I.e., shouldn't the getindex method do an `assertscalar` instead of `@allowscalar`?

Reply:
I was following the existing style in the method, I can change it if you like

Reply:
Assuming this getindex is indicative of bad scalar code, I think it should error.